global.R

```diff
@@ -68,3628 +68,16 @@ names(tables_earnings_data) <- c("X", "YAG", "subject_name", "SECTIONNAME", "sex
-    } else {
-      "#e9e9e9" # grey
-    }
-  }
-
-  # function which returns background colour based on cell value (using colour map)
-  # also takes column name as an input, which allows to get max and min
-  stylefunc <- function(value, index, name) {
-    normalized <- (value - min(yag_table_final2[name], na.rm = T)) /
-      (max(yag_table_final2[name], na.rm = T) - min(yag_table_final2[name], na.rm = T))
-    color <- orange_pal(normalized)
-    list(background = color)
-  }
-
-
-  # list giving column formatting (using style function) for single column
-  coldefs <- list(
-    colDef(
-      na = "c",
-      style = stylefunc,
-      name = "Earnings median",
-      minWidth = 30,
-      format = colFormat(prefix = "£", separators = TRUE, digits = 0)
-    )
-  )
-
-  # get names of numerical cols
-  numcols <- c("earnings_median_1YAG", "earnings_median_5YAG")
-  # replicate list to required length
-  coldefs <- rep(coldefs, length(numcols))
-  # name elements of list according to cols
-  names(coldefs) <- numcols
-
-  coldefs$earnings_change_average <- colDef(
-    style = function(value) {
-      if (value > 0 & is.na(value) == FALSE) {
-        color <- "#008000"
-      } else if (value < 0 & is.na(value) == FALSE) {
-        color <- "#e00000"
-      } else {
-        color <- "#777"
-      }
-      list(color = color, fontWeight = "bold")
-    },
-    name = "Difference in medians",
-    minWidth = 30,
-    na = "c",
-    format = colFormat(prefix = "£", separators = TRUE, digits = 0)
-  )
-
-  coldefs$SECTIONNAME.x <- colDef(name = "Industry", footer = paste("Total median for all grads in", earningsinput))
-  coldefs$SECTIONNAME.y <- colDef(name = "Industry", footer = paste("Total median for all grads in", earningsinput))
-  coldefs$count <- colDef(na = "c", minWidth = 25, name = "Number of graduates", format = colFormat(separators = TRUE))
-
-  YAG1_total <- total_earnings %>%
-    filter(YAG == 1) %>%
-    dplyr::select(earnings_median) %>%
-    mutate_at(
-      "earnings_median",
-      funs(ifelse(!is.na(as.numeric(.)), round(as.numeric(.), -2), .))
-    )
-
-  YAG5_total <- total_earnings %>%
-    filter(YAG == 5) %>%
-    dplyr::select(earnings_median) %>%
-    mutate_at(
-      "earnings_median",
-      funs(ifelse(!is.na(as.numeric(.)), round(as.numeric(.), -2), .))
-    )
-
-  coldefs$earnings_median_1YAG$footer <- paste("£", format(YAG1_total, big.mark = ",", scientific = FALSE))
-  coldefs$earnings_median_5YAG$footer <- paste("£", format(YAG5_total, big.mark = ",", scientific = FALSE))
-
-  # create table
-  earnings_table <- reactable(yag_table_final2,
-    defaultPageSize = 21, columns = coldefs, showSortable = TRUE,
-    columnGroups = list(
-      colGroup(name = "1 YAG", columns = c("SECTIONNAME.x", "earnings_median_1YAG")),
-      colGroup(name = "5 YAG", columns = c("SECTIONNAME.y", "earnings_median_5YAG"))
-    ),
-    defaultColDef = colDef(footerStyle = list(fontWeight = "bold"))
-  )
-}
-
+source('R/sankey.R')
```

Comment:
Don't need to source the scripts as they're in a folder called 'R' so shiny loads them automatically at the start. Can delete  everything from line 67 in the new script

Reply:
Otherwise this is exactly what I was thinking of though!